packages/webapp/src/containers/Finances/AddSale/RevenueTypes/useSortedRevenueTypes.jsx

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import { getRevenueTypes } from '../../saga';
+import { revenueTypesSelector } from '../../../revenueTypeSlice';
+
+export default function useSortedRevenueTypes() {
```

Comment:
@SayakaOno I will steal your question haha sorry .. Is there a reason you chose not to make this custom hook reusable?

Reply:
How do you want to make it reusable? By passing getTypes action, selector and those `revenue:${typeA.revenue_translation_key}`? I prefer to make another hook in this case.

Sorry "reusable" was very wrong word choice, but I was expecting something simpler that could be used only for these two components like this:
```
const validateUniqueTypeName = (value) => {
  const expenseNameExists = expenseTypes.some((type) => {
    return type.expense_name === value;
  });

  if (expenseNameExists) {
    return t('EXPENSE.ADD_EXPENSE.DUPLICATE_NAME');
  }
  return true;
};
```
⬇️ 
```
export const validateUniqueTypeName = (expenseTypes, value) => {
  const expenseNameExists = expenseTypes.some((type) => {
    return type.expense_name === value;
  });

  if (expenseNameExists) {
    return t('EXPENSE.ADD_EXPENSE.DUPLICATE_NAME');
  }
  return true;
};
```

Reply:
Okay sounds good haha! I figured maybe the same logic was used for expense types then maybe task types etc. But I thought I should ask!

Reply:
Ah that's true! I think it's good to have a reusable sorting function at least!